userspace/libsinsp/container.cpp

```diff
@@ -96,15 +99,51 @@ bool sinsp_container_manager::remove_inactive_containers()
 
 sinsp_container_info* sinsp_container_manager::get_container(const string& container_id)
 {
-	auto it = m_containers.find(container_id);
-	if(it != m_containers.end())
+	auto containers = m_containers.lock();
+	auto it = containers->find(container_id);
+	if(it != containers->end())
 	{
 		return &it->second;
 	}
 
 	return nullptr;
 }
 
+sinsp_container_info* sinsp_container_manager::get_or_create_container(
```

Comment:
MAJOR 
Can you  add some a UT in agent/test/sysdig/libsinsp? Then we'll move it down later.